### DISCUSSION TITLE
age verification screen

### DISCUSSION THREAD

**Discussion by pixelRealmCo** [DragonsWho/CYOACatalog#14] [Ideas]
for legal reason might be good to have an age verifaction that people click that they are above the age of 18 or whatever rules in their country.

**Comment by DragonsWho** [comment]
We need to clarify what this is even supposed to look like. I would prefer to combine this with the cookies question and do it in the least annoying way possible.

**Comment by DragonsWho** [comment]
I mean, does it have to be a huge banner blocking the page? Should it always be displayed, or if I configure the site so that by default it displays SFW content, should the banner only appear when the user switches to NSFW mode?

**Comment by pixelRealmCo** [comment]
If the whole website is nsfw, (which would be the easiest) then we woulld probably make an argument in app.tsx that if the nsfw is not pressed yes then render the age verifaction page. The page then should make sure it does not render or download any data so the only thing that is shown is that page. Would suggest checking pornhub how they do it and probably just somewhat copy that. 

If you want SFW by default you can have a popup that ask if you are okey with nsfw when pressing the nsfw button and then if you do it then it will allow NSFW content when that button is pressed or checked or whatever.

The best way is probably to have an cockie that saves if you okeyed it. ( for this specific thing you do not need an cockie aproval thing for gdpr)
